test/unit_tests/test_ProxyVerifier.cc

```diff
@@ -205,20 +205,36 @@ TEST_CASE("Test path parsing", "[ParseUrl]")
   HttpHeader header;
   SECTION("Verify a simple path can be parsed")
```

Comment:
We should parameterize and expand unit test coverage of this function. I added a similar unit test for URL parsing in ATS is here:
https://github.com/apache/trafficserver/blob/master/proxy/hdrs/unit_tests/test_URL.cc#L98

As another example, Alan worked on this here:
https://github.com/apache/trafficserver/blob/720cd0c120be28505e85db37f1f5cf489f379934/proxy/hdrs/unit_tests/test_Hdrs.cc#L43
Let's parameterize this test accordingly:

Here's the pattern:
1. Define a class with the url input and the expected header values (_scheme, uri_scheme, etc.).
1. Create a vector of tests covering various scenarios. (Path with a '/' prefix and without, URI with a query but no path, URI with no authority, etc.). Strive for completeness of coverage.
1. Loop over the vector, passing the url input to parse_url and verifying the hdr members against the expected values.

Reply:
Yes, significantly more variants are needed in the unit test. The parsing is non-trivial.